### PR TITLE
Feature/websockets ssl

### PR DESCRIPTION
**Pull request by nkoenig**
Adds SSL support to the websocket server.

To test:

1. Uncomment the `<ssl>` section in [examples/websocket.ign](https://github.com/ignitionrobotics/ign-launch/blob/feature/websockets_ssl/examples/websocket.ign). Set the `<cert_file>` to contain the full path to the [plugins/websocket_server/localhost.cert](https://github.com/ignitionrobotics/ign-launch/blob/feature/websockets_ssl/plugins/websocket_server/localhost.cert) file. Set the `<private_key_file>` to contain the full path to the [plugins/websocket_server/localhost.key](https://github.com/ignitionrobotics/ign-launch/blob/feature/websockets_ssl/plugins/websocket_server/localhost.key) file.

2. Run gazebo: `ign gazebo -v 4 -r -s shapes.sdf`

3. Run the websocket server: `ign launch examples/websocket.ign -v 4`.

4. Run: `firefox plugins/websocket_server/index.hml`. 

5. Open a new tab in firefox and go to `https://localhost:9002`. Accept the certificate.

6. Go back to the `index.hml` tab, and refresh.

### CODE REVIEW

File: plugins/websocket_server/WebsocketServer.cc [hunk 1]
```diff
@@ -239,9 +256,40 @@ bool WebsocketServer::Load(const tinyxml2::XMLElement *_elem)
...
+      return false;
+    }
+
+    // Store SSL configuration.
+    info.options = LWS_SERVER_OPTION_DO_SSL_GLOBAL_INIT;
+
+    info.ssl_cert_filepath = sslCertFile.c_str();
+    info.ssl_private_key_filepath = sslPrivateKeyFile.c_str();
+  }
+  else if (!sslCertFile.empty() || !sslPrivateKeyFile.empty())
```
Comment (AlejoAsd) [inline plugins/websocket_server/WebsocketServer.cc:284]: This should likely be
```suggestion
  else if (sslCertFile.empty() || sslPrivateKeyFile.empty())
```

File: plugins/websocket_server/index.html [hunk 2]
```diff
@@ -10,8 +10,12 @@
...
   <script>
     // Create the Connection to the Ignition websocket server
+    // This configuration uses SSL and authentication. You can disable both
+    // by replacing `wss://` with `ws://` and remove the `key` parameter.
+    // Make sure to also disable SSL and authentication in the
+    // websocket_server plugin.
     var ign = new Ignition({
-      url: 'ws://localhost:9002',
+      url: 'wss://localhost:9002',
       key: "auth_key"
```
Comment (AlejoAsd) [inline plugins/websocket_server/index.html:19]: Is this the auth key the websocket server expects to receive? Maybe this should be noted in the docs.

This repo does not have the `ign.js` code in it (and rightly so). Where can I read its source?

Comment (nkoenig) [reply plugins/websocket_server/index.html:19]: It should contain an ign.js file here: https://github.com/ignitionrobotics/ign-launch/blob/feature/websockets_ssl/plugins/websocket_server/ign.js

File: plugins/websocket_server/WebsocketServer.cc (hunk 1 above)
Comment (nkoenig) [reply plugins/websocket_server/WebsocketServer.cc]: Fixed: d98a03e33048c3507988c6b250c44cf39a917dc4

File: plugins/websocket_server/index.html (hunk 2 above)
Comment (nkoenig) [reply plugins/websocket_server/index.html:19]: I had put the authorization documentation here: https://github.com/ignitionrobotics/ign-launch/blob/feature/websockets_ssl/plugins/websocket_server/WebsocketServer.hh#L45. I think yout are looking for it in the README.md, which I'll update.

Comment (nkoenig) [reply plugins/websocket_server/index.html:19]: Updated readme: 3273662291b6177336f27e665392898840d6b07a

**Pull request merged**